mscore/pathlistdialog.cpp

```diff
@@ -41,8 +42,12 @@ PathListDialog::PathListDialog(QWidget* parent)
 
 void PathListDialog::addClicked()
       {
-      QString newPath = QFileDialog::getExistingDirectory (this, tr("Choose a directory"),
-         QString("%1/%2").arg(QDesktopServices::storageLocation(QDesktopServices::DocumentsLocation)).arg(QCoreApplication::applicationName()));
+      QString newPath = QFileDialog::getExistingDirectory(
+         this,
+         tr("Choose a directory"),
+         QString("%1/%2").arg(QDesktopServices::storageLocation(QDesktopServices::DocumentsLocation)).arg(QCoreApplication::applicationName()),
+         preferences.nativeDialogs ? QFileDialog::Options() : QFileDialog::DontUseNativeDialog
```

Comment:
Something like 
`QFileDialog::ShowDirsOnly | (preferences.nativeDialogs ? 0 : QFileDialog::DontUseNativeDialog)` works for me.